File: ByteMarket/backend/src/main/java/es/grupo04/backend/dto/UserDTO.java
package es.grupo04.backend.dto;

import java.util.List;



public record UserDTO (
    Long id,
    String name,
    Integer creationYear,
    List<String> roles,
    List<ProductBasicDTO> products,
    List<ReviewDTO> reviews,
    List<PurchaseDTO> sales,
    List<PurchaseDTO> purchases,
    String iframe,
    String image,
    Boolean hasImage
    
){
}

File: frontend/src/app/dtos/user.dto.ts
import { ProductDTO } from "./product.dto";
import { ReviewDTO } from "./review.dto";
import { PurchaseDTO } from "./purchase.dto";

export interface UserDTO {
    id: number;
    name: string;
    creationYear: number;
    roles: string[];
    products: ProductDTO[];
    reviews: ReviewDTO[];
    sales: PurchaseDTO[];
    purchases: PurchaseDTO[];
    iframe: string;
    image: string;
    hasImage: boolean;
}
